`make build-debs` is exiting with 0 status code even on failure

## Description

`make build-debs` is exiting with a 0 status code, even when it fails.

## Steps to Reproduce

1. Check out the broken version of #6855.
2. `make build-debs`

## Expected Behavior

dpkg-buildpackage failure should bubble up

## Actual Behavior

```
...
Failed to build cryptography
ERROR: Could not build wheels for cryptography, which is required to install pyproject.toml-based projects
Traceback (most recent call last):
  File "/usr/bin/dh_virtualenv", line 111, in <module>
    sys.exit(main() or 0)
  File "/usr/bin/dh_virtualenv", line 91, in main
    deploy.install_dependencies()
  File "/usr/lib/python3/dist-packages/dh_virtualenv/deployment.py", line 202, in install_dependencies
    subprocess.check_call(self.pip('-r', requirements_path))
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/srv/securedrop/debian/securedrop-app-code/opt/venvs/securedrop-app-code/bin/python', '/srv/securedrop/debian/securedrop-app-code/opt/venvs/securedrop-app-code/bin/pip', 'install', '--log=/tmp/tmpi5yh1c18', '--verbose', '--no-deps', '--no-binary=:all:', '--no-cache-dir', '-r', './requirements.txt']' returned non-zero exit status 1.
make[1]: *** [debian/rules:40: override_dh_virtualenv] Error 1
make[1]: Leaving directory '/srv/securedrop'
make: *** [debian/rules:7: binary] Error 2
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
Script done.

You can now examine or commit the log at:
/home/user/github/freedomofpress/securedrop/build/20231616-securedrop.log
user@dev ~/g/f/securedrop> echo $status
0
```

## Comments

Probably a regression from the script handling?

--- a/Makefile
+++ b/Makefile
@@ -384,7 +384,7 @@ build-debs: ## Build and test SecureDrop Debian packages
 	@echo "Building SecureDrop Debian packages..."
 	@export TERM=dumb
 	@script \
-		--command $(SDROOT)/builder/build-debs.sh \
+		--command $(SDROOT)/builder/build-debs.sh --return \
 		$(OUT)
 	@echo
 	@echo "$(SCRIPT_MESSAGE)"
@@ -396,7 +396,7 @@ build-debs-notest: ## Build SecureDrop Debian packages without running tests.
 	@echo "Building SecureDrop Debian packages, skipping tests..."
 	@export TERM=dumb
 	@NOTEST=1 script \
-		--command $(SDROOT)/builder/build-debs.sh \
+		--command $(SDROOT)/builder/build-debs.sh --return \
 		$(OUT)
 	@echo
 	@echo "$(SCRIPT_MESSAGE)"
@@ -408,7 +408,7 @@ build-debs-ossec: ## Build OSSEC Debian packages
 	@echo "Building OSSEC Debian packages"
 	@export TERM=dumb
 	@WHAT=ossec script \
-		--command $(SDROOT)/builder/build-debs.sh \
+		--command $(SDROOT)/builder/build-debs.sh --return \
 		$(OUT)
 	@echo
 	@echo "$(SCRIPT_MESSAGE)"
@@ -420,7 +420,7 @@ build-debs-ossec-notest: ## Build OSSEC Debian packages without running tests
 	@echo "Building OSSEC Debian packages, skipping tests..."
 	@export TERM=dumb
 	@NOTEST=1 WHAT=ossec script \
-	       --command $(SDROOT)/builder/build-debs.sh \
+	       --command $(SDROOT)/builder/build-debs.sh --return \
 	       $(OUT)
 	@echo
 	@echo "$(SCRIPT_MESSAGE)"

--- a/builder/image_prep.sh
+++ b/builder/image_prep.sh
@@ -26,6 +26,8 @@ if [[ $status == 42 ]]; then
     echo "Rebuilding container to update dependencies"
     $OCI_BIN rmi fpf.local/sd-server-builder
     $OCI_BIN build -t fpf.local/sd-server-builder builder/ --no-cache
+    # Reset $status and re-run the dependency check
+    status=0
     $OCI_BIN run --rm $OCI_RUN_ARGUMENTS \
         --entrypoint "/dep-check" fpf.local/sd-server-builder || status=$?
 fi

